diff --git a/android/app/build.gradle b/android/app/build.gradle
index fe8e94c..e77fa83 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -7,7 +7,7 @@ android {
         applicationId "com.proplead.tracker"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 3
+        versionCode 4
         versionName "1.2"
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         aaptOptions {
diff --git a/android/app/src/main/java/com/proplead/tracker/MainActivity.java b/android/app/src/main/java/com/proplead/tracker/MainActivity.java
index dba517e..a3c2038 100644
--- a/android/app/src/main/java/com/proplead/tracker/MainActivity.java
+++ b/android/app/src/main/java/com/proplead/tracker/MainActivity.java
@@ -1,5 +1,31 @@
 package com.proplead.tracker;
 
+import android.content.Intent;
 import com.getcapacitor.BridgeActivity;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginHandle;
+import ee.forgr.capacitor.social.login.GoogleProvider;
+import ee.forgr.capacitor.social.login.ModifiedMainActivityForSocialLoginPlugin;
+import ee.forgr.capacitor.social.login.SocialLoginPlugin;
 
-public class MainActivity extends BridgeActivity {}
+public class MainActivity extends BridgeActivity implements ModifiedMainActivityForSocialLoginPlugin {
+
+    @Override
+    public void IHaveModifiedTheMainActivityForTheUseWithSocialLoginPlugin() {}
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode >= GoogleProvider.REQUEST_AUTHORIZE_GOOGLE_MIN &&
+            requestCode < GoogleProvider.REQUEST_AUTHORIZE_GOOGLE_MAX) {
+            PluginHandle handle = getBridge() != null ? getBridge().getPlugin("SocialLogin") : null;
+            if (handle != null) {
+                Plugin plugin = handle.getInstance();
+                if (plugin instanceof SocialLoginPlugin) {
+                    ((SocialLoginPlugin) plugin).handleGoogleLoginIntent(requestCode, data);
+                }
+            }
+        }
+    }
+}
